querydsl-core/src/main/java/com/querydsl/core/support/Normalization.java

```diff
@@ -26,15 +26,13 @@
 
     // TODO simplify
     private static final Pattern FULL_OPERATION = Pattern.compile(
-            "(?<![\\d\\*/\"' ])" + "(\\b|\\(|\\s+)"  +
+            "(?<![\\d\\*/\"\\?' ])" + "(\\b|\\(|\\s+)"  +
             "(" + NUMBER + WS + "[+\\-/*]" + WS + ")+" + NUMBER + WS +
             "(?![\\d\\*/\"' ])");
 
     private static final Pattern[] OPERATIONS = {
-            Pattern.compile(NUMBER + WS + "\\*" + WS + NUMBER),
-            Pattern.compile(NUMBER + WS + "/" + WS + NUMBER),
-            Pattern.compile(NUMBER + WS + "\\+" + WS + NUMBER),
-            Pattern.compile(NUMBER + WS + "\\-" + WS + NUMBER)
+            Pattern.compile(NUMBER + WS + "(\\*|/)" + WS + NUMBER),
+            Pattern.compile(NUMBER + WS + "(\\+|\\-)" + WS + NUMBER),
```

Comment:
Can you consider using character classes?
There are only 2 options, so it can easily fit in a `java.util.regex.Pattern$BitClass` [which is available from at least 1.6](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b27/java/util/regex/Pattern.java#Pattern.BitClass), so the performance should be better than a [branch](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b27/java/util/regex/Pattern.java#4126)('|').

See [match](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b27/java/util/regex/Pattern.java#Pattern.BmpCharProperty.match%28java.util.regex.Matcher%2Cint%2Cjava.lang.CharSequence%29) and BitClass' [implementation of `isSatisfiedBy()`](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b27/java/util/regex/Pattern.java#Pattern.BitClass.isSatisfiedBy%28int%29), it then uses a simple array lookup instead of a foreach.

Also, in a character class, meta characters lose their meaning, so escaping is not necessary. (- is not a metacharacter)